### ISSUE TITLE
More anti-aliasing options

### ISSUE THREAD

**Issue by Bcompartment** [bevyengine/bevy#14657]
## What problem does this solve or what need does it fill?

With a variety of methods available, users can select best option for (visual / performance) requirements.  Some of options is improved variations of MSAA. For SSAA taking samples within a pixel, the sample positions have to be determined in some way. Although the number of ways in which this can be done is infinite, [here (wikipedia article) ](https://en.wikipedia.org/wiki/Supersampling ) is few examples.

## What solution would you like?

- [ ] CMAA2
- [ ] CSAA/EQAA
- [ ] MSAAx16
- [ ] MSAAx32
- [ ] SSAA 

Add example how to setup multiple AA algorithms working in same time like: MSAA+TAA or FXAA+TAA.

**Comment by alice-i-cecile** [comment]
I'd prefer to make it easier to extend these, and present only a handful of proven / good options in the engine itself.

**Comment by JMS55** [comment]
I agree that we don't really need additional AA algorithms (besides improved versions of TAA/TSR).

**Comment by DGriffin91** [comment]
CMAA2 would be nice since SMAA is comparatively pretty slow and FXAA is comparatively pretty low quality. I'm also a big fan of mixing TAA and things like CMAA2 (similar to Horizon Forbidden West, etc..)

**Comment by Elabajaba** [comment]
Afaik only Intel supports 16x MSAA these days (it doesn't look like any AMD or Nvidia GPUs support 16x or 32x with vulkan), and I believe both 16x and 32x MSAA were generally faked in drivers as 4x SSAA + 4xMSAA (or + 8xMSAA for 32x MSAA).

I believe CSAA/EQAA were driver level AA overrides that Nvidia/AMD removed from their driver control panels years ago.

I'm not sure if wgpu supports custom MSAA resolve, but you can improve MSAA quality with custom resolve filters (with a performance cost) https://therealmjp.github.io/posts/msaa-resolve-filters/

We also need to do a reversible tonemap before/after our MSAA resolve otherwise there's jaggies along HDR edged (https://therealmjp.github.io/posts/msaa-overview/#working-with-hdr-and-tone-mapping and https://gpuopen.com/learn/optimized-reversible-tonemapper-for-resolve/).

**Comment by TheIronLefty** [comment]
> Afaik only Intel supports 16x MSAA these days (it doesn't look like any AMD or Nvidia GPUs support 16x or 32x with vulkan), and I believe both 16x and 32x MSAA were generally faked in drivers as 4x SSAA + 4xMSAA (or + 8xMSAA for 32x MSAA).
> 
> I believe CSAA/EQAA were driver level AA overrides that Nvidia/AMD removed from their driver control panels years ago.
> 
> I'm not sure if wgpu supports custom MSAA resolve, but you can improve MSAA quality with custom resolve filters (with a performance cost) https://therealmjp.github.io/posts/msaa-resolve-filters/
> 
> We also need to do a reversible tonemap before/after our MSAA resolve otherwise there's jaggies along HDR edged (https://therealmjp.github.io/posts/msaa-overview/#working-with-hdr-and-tone-mapping and https://gpuopen.com/learn/optimized-reversible-tonemapper-for-resolve/).

EQAA is still available to AMD cards on most renderers requesting MSAA incl in DX12, you can try this on Dirt Rally 2.0, it doesn't require control panel override.
